include/class.ticket.php

```diff
@@ -3382,8 +3392,12 @@ static function create($vars, &$errors, $origin, $autorespond=true,
 
         // Assign ticket to staff or team (new ticket by staff)
         if ($vars['assignId']) {
-            $asnform = $ticket->getAssignmentForm(array('assignee' => $vars['assignId']));
-            $ticket->assign($asnform, $vars['note']);
+            $asnform = $ticket->getAssignmentForm(array(
+                        'assignee' => $vars['assignId'],
+                        'comments' => $vars['note'])
+                    );
+            $e = array();
+            $ticket->assign($asnform, $e);
```

Comment:
I somewhat agree -- I simply used the form here as a assignment request. I think we should move more towards __request based transaction__ as opposed to hash array (good example is ticket create request). This is highly important for request interception, modification (automation) and uniform data validation that is not tied to html input elements. 

My thinking is internal API, automation and pluggable mechanisms could use direct calls like `setStaffId`.